### PR TITLE
Project setup

### PR DESCRIPTION
**Pull request by martinnajjar12**
### Project setup

We've added the folders and files below:

- RSpec test file
- Rubocop
- bin/main.rb
- lib/

### CODE REVIEW

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
# Changes required 🛑 

## What a remarkable project you have built 💯! This is impressive 🤩👌🧨!

### But in order to get it approved, we need to work on some changes. Let's see...

You are off to a great start 👏 ! But before we move to the next milestone, let's make the `main.rb` file executable.

You can add this line `#!/usr/bin/env ruby` at the top of your `main.rb` to make it executable.

Your project looks absolutely fantastic. Just make these changes and apply for another code review.

Happy coding 👨‍💻 

*Need help? Shoot me a message on Slack - [@Ghosh](https://app.slack.com/client/T47CT8XPG/DRDM6LAQN/user_profile/URB1MHXK6)*
<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

File: bin/main.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# A file which has a shebang line as its first line is
+# granted execute permission.
```
Comment (kitpao) [inline bin/main.rb]: - [x] The shebangline is: `#!/usr/bin/env ruby`, this "text" was an explanation, you can delete these comments.

File: bin/main.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+# !/usr/bin/env ruby
```
Comment (kitpao) [inline bin/main.rb]: - [x] Make sure that this line is the first line of your document. Please notice that this is not a regular comment, this is a "hidden" command, so it doesn't need a space after the `#`. Rubocop won't complain about the space when it is recognized as a command (in the first line of the document)

File: bin/main.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+puts 'hello, world'
```
Comment (kitpao) [inline bin/main.rb]: - [x] Now, for the permissions for the file, if both of you have windows, you can not use the chmod command. So let's make use of a "virtual environment" with linuxbrew (this is called a pod in DevOps, you can research about it later if you want). Basically, it gives you a blank computer in the cloud with ruby and Linux previously installed, just append `gitpod.io/` to your repo URL, in this case: https://gitpod.io/github.com/edxco/tic-tac-toe/tree/project_setup

  - Log in with your github account
  - In a new tab, go to https://gitpod.io/access-control/ and give permission to write public repos
  - go back to the tab with your project in gitpod and make changes as you would in your computer, follow the instructions in the [article of pathwright](https://commandercoriander.net/blog/2013/02/16/making-a-ruby-script-executable/)
  - git add, git commit and git push :)

**Review by kitpao — CHANGES_REQUESTED** [review]
# 👋 Hello @martinnajjar12 and @edxco 

I'm sorry, your file is not yet executable,
I read in Slack that both of you have windows, so I will take that into consideration with my suggestions, please check below 🔽 

### ♻️ Status: Your project requires some additional changes ♻️
➡️ Please request another review once you are ready 📝 
___
##### 👉 If you have any doubts you can contact me in slack as @kitpao
##### 🌟  If you have feedback about my reviews I would appreciate if can leave a recommendation in LinkedIn 💌 or let me know in Slack if you see improvements to be done 😛 

🔗 __[Github](https://github.com/kitpao) | [Twitter](https://twitter.com/Kitpao1) | [LinkedIn](https://www.linkedin.com/in/kitzia-paola-vidal/) | [Medium](https://medium.com/@kitpao91)__

**Review by kitpao — CHANGES_REQUESTED** [review]
# 👋 Hello again Team,

Your files look good, however, you have linter errors and after investigating, it seems they are related to the thing preventing your file to be executed. Let's go through it:

 When trying to execute, it shows this error:
image
There is a `ruby\r`. The` \r` is called a carriage return, it is invisible to the eye but it is in the windows configuration, and you can also see it in rubocop linters:

image

That is a common error when typing in windows editors. ([You can see more info here.](https://stackoverflow.com/questions/54286602/how-to-fix-shebang-warning-in-ruby)) To get rid of it, please redo your main.rb file in a **Linux environment** (you can use a virtual machine or gitpod)
  - Delete the main file you have (since it has the "invisible" carriage returns [crlf] configured)
  - Make a new main.rb file inside bin folder
  - copy-paste: 
  ```
  #!/usr/bin/env ruby

  puts 'Hello'

  ```
  - In the terminal, in your project folder type `chmod 755 bin/main.rb`

That is it, you don't need to take out the .rb extension

To test if it works, type in the terminal `bin/main.rb`

Make Rubocop linter corrections if necessary

### ♻️ Status: Your project requires some additional changes ♻️
➡️ Please request another review once you are ready 📝 
___
##### 👉 If you have any doubts you can contact me in slack as @kitpao
##### 🌟  If you have feedback about my reviews I would appreciate if can leave a recommendation in LinkedIn 💌 or let me know in Slack if you see improvements to be done 😛 

🔗 __[Github](https://github.com/kitpao) | [Twitter](https://twitter.com/Kitpao1) | [LinkedIn](https://www.linkedin.com/in/kitzia-paola-vidal/) | [Medium](https://medium.com/@kitpao91)__

**Review by tirthajyoti-ghosh — APPROVED** [review]
# Approved ✅

## You have done an incredible job making this project 👏🧨💪! Just perfect 💯! 

### You have met all the requirements of the project. Go ahead and merge it! Onto the next project 🚀 !

Happy coding 👨‍💻 

<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Pull request merged**